src/modules/keycloak-administration/domain/keycloak-user.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Mapper } from '@automapper/core';
+import { getMapperToken } from '@automapper/nestjs';
+import { Inject, Injectable } from '@nestjs/common';
+import { KeycloakAdminClient, type UserRepresentation } from '@s3pweb/keycloak-admin-client-cjs';
+import { plainToClass } from 'class-transformer';
+import { ValidationError, validate } from 'class-validator';
+
+import { DomainError, EntityNotFoundError, KeycloakClientError } from '../../../shared/error/index.js';
+import { KeycloakAdministrationService } from './keycloak-admin-client.service.js';
+import { UserRepresentationDto } from './keycloak-client/user-representation.dto.js';
+import { UserDo } from './user.do.js';
+
+@Injectable()
+export class KeycloakUserService {
+    public constructor(
+        private readonly kcAdminService: KeycloakAdministrationService,
+        @Inject(getMapperToken()) private readonly mapper: Mapper,
+    ) {}
+
+    public async create(user: UserDo<false>, password?: string): Promise<Result<string, DomainError>> {
```

Comment:
Depends on how Keycloak is configured.
With default settings, both username and email have to be unique, otherwise Keycloak will throw an error.
You could enable duplicate emails, but you'd need to disable login with email for that to work.
There is also a setting named "Email as username" which seems to just conflate username and email, though I'm not too sure how that works behind the scenes.

Reply:
Ah I see, as far as I can see "E-Mail as username" is exactly what is wanted.

My point was: We shouldn't rely on Keycloak throwing something. I'd like us to check for duplication ourselves and gracefully handle that.

Thanks :)

Reply:
Added check before creating user